src/ipa_pc/mod.rs

```diff
@@ -111,21 +111,43 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
         for (labeled_commitment, value) in labeled_commitments.zip(values) {
             let commitment = labeled_commitment.commitment();
             combined_v += &(cur_challenge * &value);
-            combined_commitment_proj += &labeled_commitment.commitment().comm.mul(cur_challenge);
+
+            if commitment.comm.len() > 1 {
+
+                let mut comm_lc = <G::Projective as ProjectiveCurve>::zero();
+                for (i, comm_single) in commitment.comm.iter().enumerate() {
+                    let is = i * n;
+                    comm_lc += &comm_single.mul(point.pow(&[is as u64]));
+                }
```

Comment:
A slightly  more elegant way to compute the linear combination of the segment polynomials, i.e.

comm_lc = comm_single[0] + point^s * comm_single[1] + ... + point^{K*s} * comm_single[K],

is to use the Horner scheme, which usese the segement polynomials in reverse order:
We start with comm_lc = comm_single[K], and compute
comm_lc <- point*(comm_lc + comm[K-1]),
comm_lc <- point*(comm_lc + comm[K-2]),
...
comm_lc <- point*(comm_lc + comm[0]).

Reply:
If you execute the code in the if clause you perform additional (but useless) operation, with respect to the else clause. The result will be the same but you waste time and resources

Reply:
Are you refering to my comment below?

Reply:
As I understan horner scheme should be implemented in ginger-lib, not in this repo, right?